server/models/users.js

```diff
@@ -169,24 +164,12 @@ export default (sequelize, Sequelize) => {
   };
 
   User.prototype.updateProfile = async function(user) {
-    const {
-      firstName,
-      lastName,
-      gender,
-      biography,
-      mobileNumber,
-      middleName,
-      imageURL
-    } = user;
-    this.firstName = firstName || this.firstName;
-    this.middleName = middleName;
-    this.lastName = lastName || this.lastName;
-    this.gender = gender;
-    this.biography = biography;
-    this.mobileNumber = mobileNumber;
-    this.imageURL = imageURL;
-    this.save;
-    await this.reload;
+    Object.keys(user).forEach(key => {
+      this[key] = ["firstName", "lastName"].includes(key)
+        ? user[key] || this[key]
+        : user[key];
+    });
+    this.save();
     return this;
   };
```

Comment:
are you sure this action is saving?, you are not reloading it with await this.reload()?  before return this

Reply:
This already has the update, no need to reload it.